modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1107,7 +1116,7 @@ public CompletableFuture<Void> upsert(BinaryRowEx row, @Nullable InternalTransac
                         .transactionId(txo.id())
                         .enlistmentConsistencyToken(enlistmentConsistencyToken)
                         .requestType(RW_UPSERT)
-                        .timestamp(clockService.now())
+                        .timestamp(txo.startTimestamp()) // TODO replace everythere
```

Comment:
It has to do in this PR, isn't it?

Reply:
Replaced with TODO
https://issues.apache.org/jira/browse/IGNITE-23712